src/mono/wasm/runtime/invoke-cs.ts

```diff
@@ -345,12 +488,30 @@ export async function mono_wasm_get_assembly_exports(assembly: string): Promise<
             // it doesn't have the __GeneratedInitializer class
             cwraps.mono_wasm_runtime_run_module_cctor(asm);
         }
+
         endMeasure(mark, MeasuredBlock.getAssemblyExports, assembly);
     }
 
     return exportsByAssembly.get(assembly) || {};
 }
 
+async function mono_wasm_get_assembly_exports_naot(assembly: string): Promise<any> {
+    assert_bindings();
+    const result = exportsByAssembly.get(assembly);
+    if (!result) {
+        const mark = startMeasure();
+        
+        const register = (Module as any)["_" + assembly + "__GeneratedInitializer" + "__Register_"];
+        mono_assert(register, `Missing wasm export for JSExport registration function in assembly ${assembly}`);
+
+        endMeasure(mark, MeasuredBlock.getAssemblyExports, assembly);
```

Comment:
> const mark = startMeasure();

Does it make sense to measure what is effectively a dictionary access, or is it still better for consistency?

Reply:
It calls the generated register method which calls back to JS for binding individual exported methods

Reply:
Ahh, hmm, so it is this line that calls it:
```
const register = (Module as any)["_" + assembly + "__GeneratedInitializer" + "__Register_"];
```
?

Reply:
No, the actual call was missing 😀
Sorry for confusion